src/otx/v2/adapters/torch/mmengine/engine.py

```diff
@@ -649,6 +644,24 @@ def _update_codebase_config(
             codebase(str): mmX codebase framework
             task(str): mmdeploy task
         """
-        codebase = codebase if codebase is not None else self.registry.name
-        codebase_config = {"type": codebase, "task": task}
+        codebase = self.registry.name
+        mmdeploy_task_dict = {
+            TaskType.SEGMENTATION: "Segmentation",
+            TaskType.CLASSIFICATION: "Classification",
+            TaskType.DETECTION: "ObjectDetection",
+            TaskType.INSTANCE_SEGMENTATION: "InstanceSegmentation",
+        }
+        codebase_config: dict[str, str | dict] = {"type": codebase, "task": mmdeploy_task_dict[self.task]}
+        if self.task == TaskType.DETECTION:
+            # Detection task require post_processing config
+            # This should be handled model api
+            codebase_config["post_processing"] = {
+                "score_threshold": 0.05,
+                "confidence_threshold": 0.005,  # for YOLOv3
+                "iou_threshold": 0.5,
+                "max_output_boxes_per_class": 200,
+                "pre_top_k": 5000,
+                "keep_top_k": 100,
+                "background_label_id": -1,
+            }
```

Comment:
All `codebase_config` per task is going to be defined here in the future?

Reply:
Aha, I just saw your comment about model API. Do you have any idea about how to handle this kind of configurations in model API?

Reply:
I think postprocessing would be different on each model. So each model should have own codebase_config or postprocess_config to export itself. However current otxv2 do not have class for model api. Therefore I use unify config for all model, but in later phase model api should provide it's own config for export to engine. And I don't have precise plan for this